pkg/registry/generic/rest/proxy.go

```diff
@@ -131,32 +138,42 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 		return false
 	}
 
-	backendConn, err := proxy.DialURL(h.Location, h.Transport)
+	var (
+		backendConn net.Conn
+		rawResponse []byte
+		err         error
+	)
+	if h.InterceptRedirects && utilconfig.DefaultFeatureGate.StreamingProxyRedirects() {
+		backendConn, rawResponse, err = h.connectBackendWithRedirects(req)
+	} else {
+		backendConn, err = h.connectBackend(req, h.Location)
+	}
 	if err != nil {
-		h.Responder.Error(err)
+		h.Responder.Error(fmt.Errorf("error dialing backend: %v", err))
```

Comment:
Depending on what causes `connectBackend`/`connectBackendWithRedirects` to return an error, you may end up writing `error dialing backend: error dialing backend: ...`. It would be nice to avoid that if possible.